source/controlTypes/role.py

```diff
@@ -14,15 +13,36 @@
 
 @unique
 class Role(DisplayStringIntEnum):
+	"""
+	Add-on authors are recommended not to depend on values and use a `Role` directly.
+	From a string, this can be done as `controlTypes.Role[nameOfRole]` e.g. `Role["CHECKBOX"]`.
+	Although unlikely to change, if names/values changing represents a significant risk for your add-on,
+	then consider decoupling, and maintain an internal mapping of `Role` to add-on internal roles.
+
+	As add-on authors may still rely on the values, new members of `Role` should continue
+	the previous pattern of incrementing.
+	Using `enum.auto` may backfill role values. For example, creating a new role with auto value
+	68 would cause it to be included in an add-on which checks roles as follows:
+	`(66 < role) and (role < 70)`.
+

```

Comment:
Behavior that is supported, and intentionally persisted:
```python
r = ROLE_GRAPHIC  # intentionally alias use

assert isinstance(r, int)
assert r == 16
assert 15 < r < 17
```

Perhaps these should be added as unit tests and the comment point to them?

Reply:
That has been done in 0ad0d35